ncm-openstack/src/main/pan/components/openstack/schema.pan

```diff
@@ -0,0 +1,666 @@
+    'nova' : openstack_keystone_authtoken
+    'ml2' : openstack_neutron_ml2
+    'ml2_type_flat' : openstack_neutron_ml2_type_flat
+    'ml2_type_vxlan' : openstack_neutron_ml2_type_vxlan
+};
+
+
+@documentation {
+    list of horizon service configuration sections
+}
+type openstack_horizon_config = extensible {
+    @{host where is running OpenStack Keystone service}
+    'openstack_host' ? type_fqdn
+    @{If horizon is running in production (DEBUG is False), set this
+    with the list of host/domain names that the application can serve.
+    For more information see:
+    https://docs.djangoproject.com/en/dev/ref/settings/#allowed-hosts}
+    'allowed_hosts' ? string[] = list('*')
+    @{Horizon uses Djangos sessions framework for handling session data.
+    There are numerous session backends available, which are selected 
+    through the "SESSION_ENGINE" setting}
+    'session_engine' ? string = 'django.contrib.sessions.backends.cache'
+    @{External caching using an application such as memcached offers persistence
+    and shared storage, and can be very useful for small-scale deployment 
+    and/or development}
+    'caches' ? openstack_horizon_caches{}
+    'openstack_keystone_url' ? type_absoluteURI
+    @{Set this to True if running on a multi-domain model. When this is enabled, it
+    will require the user to enter the Domain name in addition to the username
+    for login}
+    'openstack_keystone_multidomain_support' ? boolean = true
+    'openstack_api_versions' ? openstack_horizon_api_versions
+    @{Overrides the default domain used when running on single-domain model
+    with Keystone V3. All entities will be created in the default domain.
+    NOTE: This value must be the name of the default domain, NOT the ID.
+    Also, you will most likely have a value in the keystone policy file like this
+    "cloud_admin": "rule:admin_required and domain_id:<your domain id>"
+    This value must be the name of the domain whose ID is specified there}
+    'openstack_keystone_default_domain' ? string = 'Default'
+    @{Configure the default role for users that you create via the dashboard}
+    'openstack_keystone_default_role' ? string = 'user'
+    'openstack_neutron_network' ? openstack_horizon_neutron_network
+    @{The timezone of the server. This should correspond with the timezone
+    of your entire OpenStack installation, and hopefully be in UTC.
+    Example: "Europe/Brussels"}
+    'time_zone' ? string
+};
+
+
+@documentation {
+Type to define OpenStack services
+Keystone, Nova, Neutron, etc
+}
+type component_openstack = {
+    include structure_component
+    'keystone' : openstack_keystone_config
+    'nova' ? openstack_nova_config
+    'nova_compute' ? openstack_nova_compute_config
+    'glance' ? openstack_glance_config
+    'neutron' ? openstack_neutron_config
+    'neutron_compute' ? openstack_neutron_common
+    'horizon' ? openstack_horizon_config
+} = dict();
```

Comment:
Is there a reason to set a default value of an empty dict()? I don't know of any other components that do

Reply:
@ned21 ok I can remove that if this is a problem